File: packages/pdf-annotator/android/src/main/java/io/capawesome/capacitorjs/plugins/pdfannotator/classes/options/OpenOptions.java
package io.capawesome.capacitorjs.plugins.pdfannotator.classes.options;

import androidx.annotation.NonNull;
import com.getcapacitor.PluginCall;
import io.capawesome.capacitorjs.plugins.pdfannotator.classes.CustomExceptions;

public class OpenOptions {

    @NonNull
    private final String path;

    public OpenOptions(@NonNull PluginCall call) throws Exception {
        this.path = OpenOptions.getPathFromCall(call);
    }

    @NonNull
    public String getPath() {
        return path;
    }

    @NonNull
    private static String getPathFromCall(@NonNull PluginCall call) throws Exception {
        String path = call.getString("path");
        if (path == null) {
            throw CustomExceptions.PATH_MISSING;
        }
        return path;
    }
}

File: packages/pdf-annotator/ios/Plugin/Classes/Options/OpenOptions.swift
import Foundation
import Capacitor

@objc public class OpenOptions: NSObject {
    let path: String

    init(_ call: CAPPluginCall) throws {
        self.path = try OpenOptions.getPathFromCall(call)
    }

    private static func getPathFromCall(_ call: CAPPluginCall) throws -> String {
        guard let path = call.getString("path") else {
            throw CustomError.pathMissing
        }
        return path
    }
}
